src/index.js

```diff
@@ -3,23 +3,23 @@ const github = require('@actions/github');
 
 /**
  * @param {string[]} changes
- * @param {string[]} projects
+ * @param {Record<string, string[]>} domains
  * @return {string[]}
  */
-const getOutput = (changes, projects) => {
-    if (changes.some((change) => change.includes('projects/common'))) {
-        return projects;
-    }
+const getOutput = (changes, domains) => {
```

Comment:
Looking into this function, chances are most of the time would be bigger than the domains which are static

I think it will make sense to switch the mapping, so that we will have path and projects
```
'{
    "projects/app": ["app"], 
    "projects/common": ["app"]
}'
```

yes, it will most probably make us write more but it is more efficient, then we can iterate over `changes` and have a single affected path on every change

```
const paths = Object.keys(domains);

cosnt output = changes.reduce((results, change) => {
    const affectedPath = paths.find((path) => path.includes(change));
    
    if (affectedPath) {
        results.push(...domains[affectedPath]);
    }

    return results;
}, []);

return [...new Set(output)];
```

What do you think?

Reply:
This was one of my earlier attempts, found it mentally harder. I think subject (project) is more important, it defines path. Putting it on first place can right away say which dependencies project has.

Reply:
Yeah I probably care too much about the computational time in case we have more than 100 files affected or stuff like that it just not gonna be so optimal but alright I get you